test/unit/lib/tasks/asset_manager_test.rb

```diff
@@ -0,0 +1,168 @@
+require "rake"
+require "test_helper"
+require "gds_api/test_helpers/asset_manager"
+
+class AssetManagerTest < ActiveSupport::TestCase
+  extend Minitest::Spec::DSL
+  include GdsApi::TestHelpers::AssetManager
+
+  let(:task) { Rake::Task["asset_manager:report_deleted_assets"] }
+
+  teardown do
+    task.reenable
+  end
+
+  context "for an asset attached only to an unpublished edition" do
+    let(:edition) { create(:unpublished_publication) }
+    let(:attachment) { create(:file_attachment_with_asset, attachable: edition) }
+
+    context "when asset is not deleted from asset manager" do
+      before do
+        attachment.attachment_data.assets.each do |asset|
+          stub_asset_manager_has_an_asset(asset.asset_manager_id, { deleted: false, file_url: asset.filename })
+        end
+      end
+
+      test "it should include the attachment in the report" do
+        assert_output(/^#{attachment.attachment_data.assets.first.filename},#{edition.public_url},#{edition.state}$/) { task.invoke }
+      end
+    end
+
+    context "when asset is deleted from asset manager" do
+      before do
+        attachment.attachment_data.assets.each do |asset|
+          stub_asset_manager_has_an_asset(asset.asset_manager_id, { deleted: true, file_url: asset.filename })
+        end
+      end
+
+      test "it should not include the attachment in the report" do
+        assert_output(/^(?!#{attachment.attachment_data.assets.first.filename},#{edition.public_url},#{edition.state}).*$/) { task.invoke }
```

Comment:
I wonder if it's worth adding a `refute_output` method somewhere to wrap around `assert_output` and make this less prone to error with the slightly complex regex. It looks like we do this in `test/unit/lib/tasks/reporting_test.rb` as well

Reply:
Good idea.  I've added a `refute_output` method and used it in both placed.